ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -547,46 +546,41 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			}
 		}
 
+		// Always remove the specified WorkersToDelete - regardless of the value of Replicas.
+		// Essentially WorkersToDelete has to be deleted to meet the expectations of the Autoscaler.
+		deletedWorkers := make(map[string]struct{})
+		deleted := struct{}{}
+		r.Log.Info("reconcilePods", "removing the pods in the scaleStrategy of", worker.GroupName)
+		for _, podsToDelete := range worker.ScaleStrategy.WorkersToDelete {
+			pod := corev1.Pod{}
+			pod.Name = podsToDelete
+			pod.Namespace = utils.GetNamespace(instance.ObjectMeta)
+			r.Log.Info("Deleting pod", "namespace", pod.Namespace, "name", pod.Name)
+			if err := r.Delete(ctx, &pod); err != nil {
+				if !errors.IsNotFound(err) {
+					return err
```

Comment:
do we log this error anywhere?

Reply:
We do not explicitly log it, but the error will be propagated to the `Reconcile` function, which will then throw an exception. Logging it explicitly is a good idea. I will add a log function.

Reply:
Updated: https://github.com/ray-project/kuberay/pull/1251/commits/d7aba0fea7fdf237f2747557576c535d5ec652b3